src/print-prescription/prescription-print-action.component.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { Button } from '@carbon/react';
+import { Printer } from '@carbon/react/icons';
+import { useTranslation } from 'react-i18next';
+import { showModal } from '@openmrs/esm-framework';
+
+type PrescriptionPrintActionProps = {
+  encounterUuid: string;
+  patientUuid: string;
+};
+
+const PrescriptionPrintAction: React.FC<PrescriptionPrintActionProps> = ({ encounterUuid, patientUuid }) => {
+  const { t } = useTranslation();
+
+  return (
+    <Button
+      renderIcon={Printer}
+      iconDescription={t('print', 'Print')}
+      onClick={() => {
+        const dispose = showModal('prescription-print-preview-modal', {
+          onClose: () => dispose(),
+          encounterUuid,
+          patientUuid,
+        });
+      }}
```

Comment:
Best to move this handler to a separate function such as:

```suggestion
const handleClick = () => {
  const dispose = showModal('prescription-print-preview-modal', {
    onClose: () => dispose(),
    encounterUuid,
    patientUuid,
  });
}
```

And then we could do:

```suggestion
      onClick={handleClick}
```